packages/client/wallets/aa/src/passkeys/PasskeysSDK.ts

```diff
@@ -0,0 +1,163 @@
+
+            logInfo("[PASSKEYS_SIGN_UP] - FINISH", { chain, walletAddress, pkpEthAddress, pkpPublicKey });
+            return { pkpEthAddress, pkpPublicKey };
+        } catch (error: any) {
+            logError("[PASSKEYS_SIGN_UP] - ERROR_SIGNING_UP", { error: errorToJSON(error), chain, walletAddress });
+            throw new PasskeySdkError(`Error signing up [${error?.name ?? ""}]`);
+        }
+    }
+
+    async encryt(chain: BlockchainIncludingTestnet, walletAddress: string, secretToEncrypt: string) {
+        try {
+            const ciphers = await this.getPasskeyCiphers(chain, walletAddress);
+            if (!this.checkPPKAlreadyMinted(ciphers)) {
+                logWarn("[PASSKEYS_ENCRYPT] - PKP not minted. Please sign up first.", { chain, walletAddress });
+                return;
+            }
+
+            if (this.checkSecretAlreadyCreated(ciphers)) {
+                logWarn("[PASSKEYS_ENCRYPT] - Secret already created. Skipping...", { chain, walletAddress });
+                return;
+            }
+            const capacityDelegationAuthSig = await this.crossmintService.getCapacityCreditsOwnerSignature();
+            const { ciphertext, dataToEncryptHash } = await this.litService.encrypt(
+                secretToEncrypt,
+                ciphers.cipher.data.pkpPublicKey!,
+                ciphers.cipher.data.pkpEthAddress!,
+                capacityDelegationAuthSig
+            );
+            await this.saveCypherText(chain, walletAddress, ciphertext, dataToEncryptHash);
+
+            logInfo("[PASSKEYS_ENCRYPT] - FINISH", { chain, walletAddress });
+            return { ciphertext, dataToEncryptHash };
+        } catch (error: any) {
+            logError("[PASSKEYS_ENCRYPT] - ERROR_ENCRYPTING", { error: errorToJSON(error), chain, walletAddress });
+            throw new PasskeySdkError(`Error encrypting message [${error?.name ?? ""}]`);
+        }
+    }
+
+    async decrypt(chain: BlockchainIncludingTestnet, walletAddress: string) {
+        try {
+            const ciphers = await this.getPasskeyCiphers(chain, walletAddress);
+            if (!this.checkPPKAlreadyMinted(ciphers)) {
+                logWarn("[PASSKEYS_DECRYPT] - PKP not minted. Please sign up first.", { chain, walletAddress });
+                return;
+            }
+
+            if (!this.checkSecretAlreadyCreated(ciphers)) {
+                logWarn("[PASSKEYS_DECRYPT] - Secret not created. Please create one first.", { chain, walletAddress });
+                return;
+            }
+            const capacityDelegationAuthSig = await this.crossmintService.getCapacityCreditsOwnerSignature();
+            const decryptedString = await this.litService.decrypt(
+                ciphers.cipher.data.pkpPublicKey!,
+                ciphers.cipher.data.pkpEthAddress!,
+                ciphers.cipher.data.cipherText!,
+                ciphers.cipher.data.dataToEncryptHash!,
```

Comment:
```suggestion
                ciphers.cipher.data.pkpPublicKey,
                ciphers.cipher.data.pkpEthAddress,
                ciphers.cipher.data.cipherText,
                ciphers.cipher.data.dataToEncryptHash,
```

Reply:
same here, I'm running:
```            
            if (!this.checkPPKAlreadyMinted(ciphers)) {
                logWarn("[PASSKEYS_DECRYPT] - PKP not minted. Please sign up first.", { chain, walletAddress });
                return;
            }

            if (!this.checkSecretAlreadyCreated(ciphers)) {
                logWarn("[PASSKEYS_DECRYPT] - Secret not created. Please create one first.", { chain, walletAddress });
                return;
            }
```
before this step, but typescript doesn't recognize that they are not null